### DISCUSSION TITLE
Embed code snippets into README

### DISCUSSION THREAD

**Discussion by nifadyev** [satwikkansal/wtfpython#340] [Ideas]
Hey @satwikkansal . Current contribution workflow contains a lot of duplication. For example, each translation duplicates code snippets and translates only comments. What do you think about using [embedding code](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-a-permanent-link-to-a-code-snippet#linking-to-markdown) [Github Action](https://github.com/marketplace/actions/markdown-embed-code-from-file)?
The goal is to store, test and analyze code snippets in one place, and reduce duplicity. Of course, it will probably brake current interactive notebooks or CLI, but it will bring more benefits than drawbacks. CI checks and validations, running snippets on various Python versions, automatic PR checks and so on

**Comment by satwikkansal** [comment]
Hey @nifadyev looks like a nice idea to try. 

> The goal is to store, test, and analyze code snippets in one place, and reduce duplicity.
Makes sense. I'd still prefer 1 file per example, so we don't have to change the line numbers of all other embeddings if we add a new line or remove something.

> Of course, it will probably brake current interactive notebooks or CLI, but it will bring more benefits than drawbacks.

It shouldn't, I think it's still possible to maintain interactive notebooks as they are generated programmatically from the readme as well.  Check: https://github.com/satwikkansal/wtfpython/tree/master/irrelevant

The code'll need to be adjusted slightly, but it'll work.

**Reply by nifadyev** [reply]
Good to hear. I will create an issue for adding embedded code snippets. And another one for checking and refactoring notebook generator
